### ISSUE TITLE
想把特征维度由2048维降低点，比如，256维或者128维，怎么办

### ISSUE THREAD

**Issue by szh91** [layumi/Person_reID_baseline_pytorch#231]

**Comment by layumi** [comment]
1.现在用的是 512 dim，不是2048。你可以看看怎么学的。
2. 你可以修改 模型 加一层 256 或者 128在分类层前。

**Comment by szh91** [comment]
可以说具体点吗，不知道怎么修改？是在model.py的ClassBlock类里面还是在PCB类里面？

**Comment by layumi** [comment]
对。在classifier的module 中。
https://github.com/layumi/Person_reID_baseline_pytorch/blob/master/model.py#L35

**Comment by szh91** [comment]
if linear:
     add_block += [nn.Linear(input_dim, 256)]
     add_block += [nn.Linear(256, num_bottleneck)]
是这样的吗？

**Comment by szh91** [comment]
还是这样呢
 add_block = []
        if linear:
            add_block += [nn.Linear(input_dim, 256)]
           # add_block += [nn.Linear(256, num_bottleneck)]
        else:
            num_bottleneck = input_dim
        if bnorm:
            #add_block += [nn.BatchNorm1d(num_bottleneck)]
            add_block += [nn.BatchNorm1d(256)]

        if relu:
            add_block += [nn.LeakyReLU(0.1)]
        if droprate>0:
            add_block += [nn.Dropout(p=droprate)]
        add_block = nn.Sequential(*add_block)
        add_block.apply(weights_init_kaiming)

        classifier = []
       #classifier += [nn.Linear(num_bottleneck, class_num)]
        classifier += [nn.Linear(256, class_num)]

**Comment by layumi** [comment]
嗯。只要改 num_bottleneck 就行。

**Comment by szh91** [comment]
师兄好，测试时候，提的怎么还是2048维特征，也要修改吗
